### PR TITLE
DAT-3639 | Be able to edit Wikia Description via CC editor

### PR DESCRIPTION
**Pull request by dianafa**
DO NOT MERGE!

https://wikia-inc.atlassian.net/browse/DAT-3639

### CODE REVIEW

**Comment by vforge** [comment]
```
/var/lib/jenkins/workspace/Mercury-PR-Checks/front/main/app/components/curated-content-editor-wikia-description.js
  23:22  error  Opening curly brace does not appear on the same line as controlling statement  brace-style
  31:21  error  Opening curly brace does not appear on the same line as controlling statement  brace-style

/var/lib/jenkins/workspace/Mercury-PR-Checks/front/main/app/components/curated-content-editor.js
  54:5   error  Unexpected console statement  no-console
  54:17  error  Strings must use singlequote  quotes

/var/lib/jenkins/workspace/Mercury-PR-Checks/front/main/app/models/curated-content-editor.js
  114:5  error  Unnecessarily quoted property `featured` found        quote-props
  115:5  error  Unnecessarily quoted property `optional` found        quote-props
  116:5  error  Unnecessarily quoted property `curated` found         quote-props
  117:5  error  Unnecessarily quoted property `community_data` found  quote-props
```

File: front/main/app/components/curated-content-editor-wikia-description.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import TrackClickMixin from '../mixins/track-click';
+import IEIFrameFocusFixMixin from '../mixins/ieiframe-focus-fix';
+
+export default Ember.Component.extend(
+	TrackClickMixin,
+	IEIFrameFocusFixMixin,
+	{
+		classNames: ['curated-content-editor-wikia-description', 'curated-content-editor-block'],
+		debounceDuration: 300,
```
Comment (rogatty) [inline front/main/app/components/curated-content-editor-wikia-description.js]: If you don't use it - remove it.

File: front/main/app/components/curated-content-editor-wikia-description.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		spinnerOverlay: false,
```
Comment (rogatty) [inline front/main/app/components/curated-content-editor-wikia-description.js]: If you don't use it - remove it.

File: front/main/app/components/curated-content-editor-wikia-description.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+		isTooltipVisible: false,
```
Comment (rogatty) [inline front/main/app/components/curated-content-editor-wikia-description.js]: If you don't use it - remove it.

File: front/main/app/models/curated-content-editor.js [hunk 4]
```diff
@@ -89,7 +90,7 @@ CuratedContentEditorModel.reopenClass({
 				},
 				success: (data) => {
 					if (Ember.isArray(data.data)) {
-						resolve(CuratedContentEditorModel.sanitize(data.data));
+						resolve(CuratedContentEditorModel.sanitize(data));
```
Comment (rogatty) [inline front/main/app/models/curated-content-editor.js]: Are you sure about this change? One line above there is a check for `data.data`

File: front/main/app/models/curated-content-editor.js [hunk 5]
```diff
@@ -130,17 +132,22 @@ CuratedContentEditorModel.reopenClass({
...
+				community_data: 'true'
 			};
 
-		if (rawData.length) {
-			rawData.forEach((section) => {
+		if (rawData.data.length) {
+			rawData.data.forEach((section) => {
 				if (section.featured === 'true') {
 					featured = section;
+				} if (section.community_data === 'true') {
```
Comment (rogatty) [inline front/main/app/models/curated-content-editor.js]: Is it ever true? It seems like it's located in `rawData.community_data`

File: front/main/app/templates/components/curated-content-editor-wikia-description.hbs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<header>{{i18n 'app.curated-content-editor-wikia-description'}}</header>
+<a class="curated-content-editor-info" {{action 'showHelp'}}>{{svg 'info' class='pointer-events-none' viewBox='0 0 18 18'}}</a>
+<div class="block-description">{{i18n 'app.curated-content-editor-wikia-description-description'}}</div>
+<div class="curated-content-input-wrapper">
+	<div class="input-container {{if labelErrorMessage 'error'}}">
```
Comment (rogatty) [inline front/main/app/templates/components/curated-content-editor-wikia-description.hbs]: `labelErrorMessage` isn't set anywhere.

File: front/main/app/templates/components/curated-content-editor-wikia-description.hbs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+		{{input
+		value=model.description
```
Comment (rogatty) [inline front/main/app/templates/components/curated-content-editor-wikia-description.hbs]: Please add indentation.

File: front/main/app/templates/components/curated-content-editor-wikia-description.hbs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+			name='label'
+			focus-out='setLabelFocusedOut'
+			focus-in='setLabelFocusedIn'
+		}}
+		<label class="floating-label {{if isLabelActive 'active'}}" for="label">{{i18n 'app.curated-content-editor-enter-wikia-description'}}</label>
+	</div>
+</div>
+{{#modal-dialog isVisible=isHelpVisible}}
+	{{i18n 'app.curated-content-editor-wikia-description-help'}}
+{{/modal-dialog}}
```
Comment (rogatty) [inline front/main/app/templates/components/curated-content-editor-wikia-description.hbs]: See that red thing?

**Comment by SebastianMarzjan** [comment]
Please resolve conflicts

**Comment by hakubo** [comment]
Lgtm

**Comment by hakubo** [comment]
side note: It probably would be good to add this https://github.com/kategengler/ember-feature-flags somewhere in the future to standardise how we use feature flags

**Comment by SebastianMarzjan** [comment]
@hakubo Yup, a great idea. It it currently on the backlog somewhere?

**Comment by rogatty** [comment]
LGTM

@SebastianMarzjan https://wikia-inc.atlassian.net/browse/XW-641

**Comment by SebastianMarzjan** [comment]
Thanks!

**Pull request merged**